PhysicsTools/NanoAOD/python/nano_cff.py

```diff
@@ -101,8 +103,10 @@ def nanoAOD_customizeMC(process):
 _80x_sequenceMC = nanoSequenceMC.copy()
 _80x_sequenceMC.remove(genSubJetAK8Table)
 _80x_sequenceMC.remove(genJetFlavourTable)
+_80x_sequenceMC.remove(ttbarTablesMC)
 _80x_sequenceMC.insert(-1,genJetFlavourAssociation)
 _80x_sequenceMC.insert(-1,genJetFlavourTable)
+_80x_sequenceMC.insert(-1,ttbarTablesMC)
```

Comment:
here things seem a bit messed up (probably already in the genJetFlavour stuff).

The schema for the sequences is to disinguish a PAT related sequence  to be run first and a sequence with only Table producers last. In between there are tools such as the relinker that need to act on "final PAT products" before tables are made.

Reply:
But the problem is that the modules **matchGenBHadron** and **matchGenCHadron** need to run after the **genJetFlavourAssociation**.  So just splitting into table producers and the rest would not help.

If the **genJetFlavourAssociation** would be inserted before the first module requiring it instead of appended to the end, this would not be needed for other modules.

Reply:
well the genJetFlavourAssociation should also not be there with the Tables, one case use different ways of inserting, not necessarily "-1"